### DISCUSSION TITLE
Add Custom Metadata/Properties to Applications

### DISCUSSION THREAD

**Discussion by rkesters** [DependencyTrack/dependency-track#2712] [Q&A]
We would like to use DT in our FLOSS approval process, hence we would like to record if the item had been approved and mitigations required for use.  Is there away to add this to the data store without a code change to the API Server? Understand we may need to update the UI.

**Comment by syalioune** [comment]
Hello @rkesters 
If by applications, you mean DT projects : they can have custom tags and properties.
![image](https://github.com/DependencyTrack/dependency-track/assets/6144741/7228ddbf-5c5a-4f05-861d-618bd1d8be77)

If by applications, you mean OSS components that your company use, there are no such things to the best of my knowledge. However you can maybe leverage the policy engine and create rules with all of the components that has been approved by your company. This way, you'll have a report of all the projects containing non vetted libs. This is indeed possible if we are talking about a small number of components. For a large set, this can prove cumbersome unless automated.

**Reply by rkesters** [reply]
Thank you, I have submitted a PR to add properties.
